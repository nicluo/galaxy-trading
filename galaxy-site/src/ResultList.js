import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const ResultList = ({statements}) => {
  const cards = statements.map((s) => {
    const classes = classNames('list-group-item', {
      'list-group-item-danger': s.type === 'error',
      'list-group-item-success': s.type === 'success'
    });
    return (
      <div className="card my-2" key={s.id}>
        <ul className="list-group list-group-flush">
          <li className="list-group-item">
            <strong className="h4">
              {s.query}
            </strong>
          </li>
          <li className={classes}>
            {s.message}
          </li>
        </ul>
      </div>
    );
  });

  return (
    <div className="container my-4">
      { cards.length > 0 ? cards : DefaultMessage }
    </div>
  );
};

const DefaultMessage = (
  <div className="card my-2">
    <ul className="list-group list-group-flush">
      <li className="list-group-item">
        <strong className="h4">
          Welcome!
        </strong>
      </li>
      <li className="list-group-item list-group-item-info">
        Submit a query to get started!
      </li>
    </ul>
  </div>
);

ResultList.propTypes = {
  statements: PropTypes.array.isRequired
};

export default ResultList;

