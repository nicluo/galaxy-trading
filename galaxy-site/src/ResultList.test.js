import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ResultList from './ResultList';

describe('ResultList', () => {
  it('renders <ResultList /> component', () => {
    const wrapper = shallow(<ResultList statements={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders <ResultList /> component with item', () => {
    const statements = [{
      id: 1,
      type: 'success',
      query: 'tegj is L',
      message: 'tegj aliased to L',
    }];
    const wrapper = shallow(<ResultList statements={statements} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders <ResultList /> component with error item', () => {
    const statements = [{
      id: 1,
      type: 'error',
      query: 'bad input',
      message: 'I dont understand what you are talking about',
    }];
    const wrapper = shallow(<ResultList statements={statements} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});