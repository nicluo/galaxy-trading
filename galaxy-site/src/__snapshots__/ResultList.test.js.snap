// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultList renders <ResultList /> component 1`] = `
<div
  className="container my-4"
>
  <div
    className="card my-2"
  >
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <strong
          className="h4"
        >
          Welcome!
        </strong>
      </li>
      <li
        className="list-group-item list-group-item-info"
      >
        Submit a query to get started!
      </li>
    </ul>
  </div>
</div>
`;

exports[`ResultList renders <ResultList /> component with error item 1`] = `
<div
  className="container my-4"
>
  <div
    className="card my-2"
    key="1"
  >
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <strong
          className="h4"
        >
          bad input
        </strong>
      </li>
      <li
        className="list-group-item list-group-item-danger"
      >
        I dont understand what you are talking about
      </li>
    </ul>
  </div>
</div>
`;

exports[`ResultList renders <ResultList /> component with item 1`] = `
<div
  className="container my-4"
>
  <div
    className="card my-2"
    key="1"
  >
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <strong
          className="h4"
        >
          tegj is L
        </strong>
      </li>
      <li
        className="list-group-item list-group-item-success"
      >
        tegj aliased to L
      </li>
    </ul>
  </div>
</div>
`;
